filter_test.go

```diff
@@ -1586,6 +1590,9 @@ func TestFilterFlowerAddDel(t *testing.T) {
 	if filter.EncDestPort != flower.EncDestPort {
 		t.Fatalf("Flower EncDestPort doesn't match")
 	}
+	if flower.IPProto != nil && *filter.IPProto != *flower.IPProto {
```

Comment:
@e0ne i believe we expect flower.IPProto to not be nil  else we should fail.

so this condition should be:

```
if flower.IPProto == nil || *flower.IPProto != filter.IPProto {
   t.Fatalf("Flower IPProto doesn't match")
}
```